src/player.c

```diff
@@ -350,51 +356,61 @@ BarPlayerMode BarPlayerGetMode (player_t * const player) {
 static int play (player_t * const player) {
 	assert (player != NULL);
 
-	AVPacket pkt;
+	const unsigned int minQueueSize = 256;
+	pthread_mutex_lock (&player->aoplay_lock);
+	player->forcePause = true;
+	player->queueDone = false;
+	player->queueSize = 0;
+	pthread_mutex_unlock (&player->aoplay_lock);
+	pthread_t aoplaythread;
+	pthread_create (&aoplaythread, NULL, BarAoPlayThread, player);
+
+	AVPacket *pkt = NULL;
 	AVCodecContext * const cctx = player->cctx;
-	av_init_packet (&pkt);
-	pkt.data = NULL;
-	pkt.size = 0;
 
 	AVFrame *frame = NULL, *filteredFrame = NULL;
 	frame = av_frame_alloc ();
 	assert (frame != NULL);
-	filteredFrame = av_frame_alloc ();
-	assert (filteredFrame != NULL);
 
 	enum { FILL, DRAIN, DONE } drainMode = FILL;
 	int ret = 0;
 	while (!shouldQuit (player) && drainMode != DONE) {
+		/* alloc queue element for AVPacket */
+		struct _QUEUE_ELEMT * pkt_element =
+				(struct _QUEUE_ELEMT *) calloc (1, sizeof (struct _QUEUE_ELEMT));
```

Comment:
Usually one would use a pre-allocated ring buffer here to avoid a) frequent allocations and b) unbounded memory usage (stailq size is not limited).